loglist/logfilter.go

```diff
@@ -39,22 +39,35 @@ func (ll *LogList) ActiveLogs() LogList {
 }
 
 // Compatible creates a new LogList containing only logs of original LogList
-// compatible with cert-chain provided.
-// Cert-chain is expected to be full: ending with CA-cert.
+// compatible with cert-chain provided. Treat Logs as compatible when their
+// root info is missing.
+// Cert-chain is expected to be full or empty: ending with CA-cert or have
+// len 0.
 func (ll *LogList) Compatible(rootedChain []*x509.Certificate, roots LogRoots) LogList {
 	var compatible LogList
 	// Keep all the operators.
 	compatible.Operators = ll.Operators
-	if len(rootedChain) == 0 || !rootedChain[len(rootedChain)-1].IsCA {
+
+	// When chain info is not available, collect Logs with no root info as
+	// compatible.
```

Comment:
What's your use case for this?  If you just want a way to extract those logs of `ll` that have no roots, I'd suggest adding a new entrypoint rather than overloading a special case of the input arguments for this function.

Reply:
The use case: no or partial info on LogRoots. (before roots are collected)

In the absence of that info, we cannot neither build the chain (using ValidateChain method) nor filter out incompatible Logs.

Reply:
I can see why you might not (yet) have roots for all the logs, but what's the use case for when the caller passes in `nil` (or `[]*x509.Certificate{}`) as the first argument?  Does it boil down to something that's actually more like `ll.FindLogsWithoutRoots(roots)` ?

Reply:
It goes exactly to this logic for that case. And I was picking between options: embed it into PR or wrap around calls. 
Do you have preference for taking it out?

Reply:
I don't really have a preference, I was just trying to understand how this will be used.  Are there going to be calls like `missing := ll.Compatible(nil, curRoots)` ?

Reply:
Rather:
filtered := ll.Compatible(chain, curRoots) 
where chain might be nil.

Reply:
OK, fair enough then.